From b975ef82060e8cc33a06bba2ce2da9d4b34d4d4f Mon Sep 17 00:00:00 2001
From: Julian Smith <julian.smith@artifex.com>
Date: Mon, 8 Sep 2025 05:48:34 +0100
Subject: [PATCH] scripts/wrap/parse.py:get_args(): improve caching of results.

We now use Cursor directly in our cache's dict key, insead of
(cursor.location.file, cursor.location.line). This might avoid reported bug
with clang-21 where cursor.location.file is not hashable.
---
 scripts/wrap/parse.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/wrap/parse.py b/scripts/wrap/parse.py
index 1954101af..578d76b71 100644
--- a/scripts/wrap/parse.py
+++ b/scripts/wrap/parse.py
@@ -509,7 +509,7 @@ def get_args( tu, cursor, include_fz_context=False, skip_first_alt=False, verbos
     #
     if verbose:
         jlib.log( '## Looking at args of {cursor.spelling=}')
-    key = tu, cursor.location.file, cursor.location.line, include_fz_context, skip_first_alt
+    key = tu, cursor, include_fz_context, skip_first_alt
     ret = get_args_cache.get( key)
     if not verbose and state.state_.show_details(cursor.spelling):
         verbose = True
